icacls_work.rb

```diff
@@ -1,36 +1,38 @@
-class Rights
+class Permission
 
   attr_reader :path, :name
   def initialize(path, name)
     @path = path
     @name = name
   end
 
-  def get_res_string(name_file)
-    system("cd #{@path} && icacls #{@name} > #{name_file}.txt")
-    written_file = File.open("#{@path}\\#{name_file}.txt")
-    written_file.each do |line|
-      return line if line.include?'Users'
+  def users_string
+    line = `cd #{@path} && icacls #{@name}`
+    line_users = ''
+    (line.index('Users')..line.size).each do |number|
+      line_users += line[number]
+      break if line[number] == "\n"
     end
+    line_users
   end
 
-  def print_rights(rights_line)
-    res_string = 'Users have '
-    case
-    when rights_line.include?('(F)') then res_string += 'full access'
-    when rights_line.include?('(N)') then res_string += 'no access'
-    when rights_line.include?('(M)') then res_string += 'modify access'
-    when rights_line.include?('(RX)') then res_string += 'read-and-execute access'
-    when rights_line.include?('(R)') then res_string += 'read access'
-    when rights_line.include?('(W)') then res_string += 'write access'
-    when rights_line.include?('(D)') then res_string += 'delete access'
-    else res_string += '?????'
+  def print_permissions(permission_line)
+    result_string = 'Users have '
+    permission = { '(F)' => 'full access',
+                   '(N)' => 'no access',
+                   '(M)' => 'modify access',
+                   '(RX)' => 'read-and-execute access',
```

Comment:
Please, move hash out of the method. Beginning of the file probably the best place for it. Also, it should be a constant if you don't change it and don't forget about 'freeze' at the end ;)

Reply:
Btw, the best way is to create a constant in my class, I clearly understand you?

Reply:
e.g. MYCONST = ['a', 'b', 'c'].freeze
In Ruby, constants are mutable. Freeze allows you to make them immutable

Reply:
constants are mutable? Really? :) Ohh, great. I take it into account.

Reply:
I know it's weird.. better to read about it to understand :)

Reply:
RuboCop make me use word 'freeze' :) I even don't know about it. Thank you, done